lib/zig.h

```diff
@@ -1289,41 +1294,50 @@ static inline zig_i128 zig_not_i128(zig_i128 val, zig_u8 bits) {
 }
 
 static inline zig_u128 zig_shr_u128(zig_u128 lhs, zig_u8 rhs) {
+    if (rhs == zig_as_u8(0)) return lhs;
     if (rhs >= zig_as_u8(64)) return (zig_u128){ .hi = lhs.hi << (rhs - zig_as_u8(64)), .lo = zig_minInt_u64 };
```

Comment:
There are also problems with the other two paths in this (and the other shift functions), I have fixes for them here: https://github.com/ziglang/zig/pull/14002/files#diff-1109733b40c9c64563e968f39af65b92aaeae9d656d201a0d23b4cf0e8857cf9

Reply:
Rebased so this PR now only contains a patch on top of those changes.